### DISCUSSION TITLE
Chrome DevTools protocol (Network events)

### DISCUSSION THREAD

**Discussion by deakdomi** [php-webdriver/php-webdriver#878] [Q&A]
Hi!

I can't find the way how can I get the request headers with the Chrome DevTools protocol?

I've managed to reach it with **Network.requestWillBeSent** event in javascript but I need this php solution.

Thanks for your help!

**Comment by OndraM** [comment]
Hi @deakdomi , Chrome DevTools protocol usage is covered in our wiki:

https://github.com/php-webdriver/php-webdriver/wiki/Chrome#chrome-devtools-protocol-cdp

**Reply by deakdomi** [reply]
Hi!

Yes, I know, thanks, but the following code doesn't work for some reason:

$devTools = new ChromeDevToolsDriver($driver);
$devTools->execute('Network.enable');
$headers = $devTools->execute('Network.**requestWillBeSent**');

Error: unhandled inspector error: {"code":-32601,"message":"'Network.requestWillBeSent' wasn't found"}

The described examples in the wiki all Methods. The Events might need to be treated differently. Don't know!

Thanks again!

**Reply by OndraM** [reply]
Yes, regarding Events, its unfortunately not possible to bind to them using php-webdriver, as we don't connect to CDP using WebSockets.

See my response here: https://github.com/php-webdriver/php-webdriver/discussions/880#discussioncomment-596598

**Comment by deakdomi** [comment]
Thanks for your answer!

Ondřej Machulda ***@***.***> ezt írta (időpont: 2021. ápr.
14., Sze 12:05):
